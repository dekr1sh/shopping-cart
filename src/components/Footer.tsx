import { Link } from "react-router-dom";
import { Instagram, Twitter, Facebook, Youtube } from "lucide-react";

export default function Footer() {
  return (
    <footer className="bg-muted py-8 mt-auto">
      <div className="mx-auto px-4">
        <div className="grid grid-cols-1 md:grid-cols-4 gap-8">
          <div>
            <h3 className="font-bold text-lg mb-4">DripKicks</h3>
            <p className="text-muted-foreground">
              Your premium destination for the latest and greatest sneakers.
            </p>
          </div>
          <div>
            <h3 className="font-bold text-lg mb-4">Shop</h3>
            <ul className="space-y-2">
              <li>
                <Link to="/shop" className="text-muted-foreground hover:text-foreground">
                  All Sneakers
                </Link>
              </li>
              <li>
                <Link to="/shop" className="text-muted-foreground hover:text-foreground">
                  New Arrivals
                </Link>
              </li>
              <li>
                <Link to="/shop" className="text-muted-foreground hover:text-foreground">
                  Best Sellers
                </Link>
              </li>
              <li>
                <Link to="/shop" className="text-muted-foreground hover:text-foreground">
                  Sale
                </Link>
              </li>
            </ul>
          </div>
          <div>
            <h3 className="font-bold text-lg mb-4">Support</h3>
            <ul className="space-y-2">
              <li>
                <Link to="#" className="text-muted-foreground hover:text-foreground">
                  Contact Us
                </Link>
              </li>
              <li>
                <Link to="#" className="text-muted-foreground hover:text-foreground">
                  FAQs
                </Link>
              </li>
              <li>
                <Link to="#" className="text-muted-foreground hover:text-foreground">
                  Shipping
                </Link>
              </li>
              <li>
                <Link to="#" className="text-muted-foreground hover:text-foreground">
                  Returns
                </Link>
              </li>
            </ul>
          </div>
          <div>
            <h3 className="font-bold text-lg mb-4">Connect</h3>
            <div className="flex space-x-4">
              <Link to="#" className="text-muted-foreground hover:text-foreground">
                <Instagram size={24} />
              </Link>
              <Link to="#" className="text-muted-foreground hover:text-foreground">
                <Twitter size={24} />
              </Link>
              <Link to="#" className="text-muted-foreground hover:text-foreground">
                <Facebook size={24} />
              </Link>
              <Link to="#" className="text-muted-foreground hover:text-foreground">
                <Youtube size={24} />
              </Link>
            </div>
          </div>
        </div>
        <div className="border-t mt-8 pt-8 text-center text-muted-foreground">
          <p>&copy; {new Date().getFullYear()} DripKicks. All rights reserved.</p>
        </div>
      </div>
    </footer>
  );
}