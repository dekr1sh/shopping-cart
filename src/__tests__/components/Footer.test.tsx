import Footer from "@/components/Footer";
import { render } from "@/__tests__/test-utils"; 
import { describe, it, expect } from "vitest";

describe("Footer", () => {
  it("renders correctly", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
