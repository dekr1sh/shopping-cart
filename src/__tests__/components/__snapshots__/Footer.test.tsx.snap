// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer > renders correctly 1`] = `
<div>
  <footer
    class="bg-muted py-8 mt-auto"
  >
    <div
      class="mx-auto px-4"
    >
      <div
        class="grid grid-cols-1 md:grid-cols-4 gap-8"
      >
        <div>
          <h3
            class="font-bold text-lg mb-4"
          >
            DripKicks
          </h3>
          <p
            class="text-muted-foreground"
          >
            Your premium destination for the latest and greatest sneakers.
          </p>
        </div>
        <div>
          <h3
            class="font-bold text-lg mb-4"
          >
            Shop
          </h3>
          <ul
            class="space-y-2"
          >
            <li>
              <a
                class="text-muted-foreground hover:text-foreground"
                data-discover="true"
                href="/shop"
              >
                All Sneakers
              </a>
            </li>
            <li>
              <a
                class="text-muted-foreground hover:text-foreground"
                data-discover="true"
                href="/shop"
              >
                New Arrivals
              </a>
            </li>
            <li>
              <a
                class="text-muted-foreground hover:text-foreground"
                data-discover="true"
                href="/shop"
              >
                Best Sellers
              </a>
            </li>
            <li>
              <a
                class="text-muted-foreground hover:text-foreground"
                data-discover="true"
                href="/shop"
              >
                Sale
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h3
            class="font-bold text-lg mb-4"
          >
            Support
          </h3>
          <ul
            class="space-y-2"
          >
            <li>
              <a
                class="text-muted-foreground hover:text-foreground"
                data-discover="true"
                href="/"
              >
                Contact Us
              </a>
            </li>
            <li>
              <a
                class="text-muted-foreground hover:text-foreground"
                data-discover="true"
                href="/"
              >
                FAQs
              </a>
            </li>
            <li>
              <a
                class="text-muted-foreground hover:text-foreground"
                data-discover="true"
                href="/"
              >
                Shipping
              </a>
            </li>
            <li>
              <a
                class="text-muted-foreground hover:text-foreground"
                data-discover="true"
                href="/"
              >
                Returns
              </a>
            </li>
          </ul>
        </div>
        <div>
          <h3
            class="font-bold text-lg mb-4"
          >
            Connect
          </h3>
          <div
            class="flex space-x-4"
          >
            <a
              class="text-muted-foreground hover:text-foreground"
              data-discover="true"
              href="/"
            >
              <svg
                class="lucide lucide-instagram"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <rect
                  height="20"
                  rx="5"
                  ry="5"
                  width="20"
                  x="2"
                  y="2"
                />
                <path
                  d="M16 11.37A4 4 0 1 1 12.63 8 4 4 0 0 1 16 11.37z"
                />
                <line
                  x1="17.5"
                  x2="17.51"
                  y1="6.5"
                  y2="6.5"
                />
              </svg>
            </a>
            <a
              class="text-muted-foreground hover:text-foreground"
              data-discover="true"
              href="/"
            >
              <svg
                class="lucide lucide-twitter"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M22 4s-.7 2.1-2 3.4c1.6 10-9.4 17.3-18 11.6 2.2.1 4.4-.6 6-2C3 15.5.5 9.6 3 5c2.2 2.6 5.6 4.1 9 4-.9-4.2 4-6.6 7-3.8 1.1 0 3-1.2 3-1.2z"
                />
              </svg>
            </a>
            <a
              class="text-muted-foreground hover:text-foreground"
              data-discover="true"
              href="/"
            >
              <svg
                class="lucide lucide-facebook"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M18 2h-3a5 5 0 0 0-5 5v3H7v4h3v8h4v-8h3l1-4h-4V7a1 1 0 0 1 1-1h3z"
                />
              </svg>
            </a>
            <a
              class="text-muted-foreground hover:text-foreground"
              data-discover="true"
              href="/"
            >
              <svg
                class="lucide lucide-youtube"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M2.5 17a24.12 24.12 0 0 1 0-10 2 2 0 0 1 1.4-1.4 49.56 49.56 0 0 1 16.2 0A2 2 0 0 1 21.5 7a24.12 24.12 0 0 1 0 10 2 2 0 0 1-1.4 1.4 49.55 49.55 0 0 1-16.2 0A2 2 0 0 1 2.5 17"
                />
                <path
                  d="m10 15 5-3-5-3z"
                />
              </svg>
            </a>
          </div>
        </div>
      </div>
      <div
        class="border-t mt-8 pt-8 text-center text-muted-foreground"
      >
        <p>
          Â© 
          2025
           DripKicks. All rights reserved.
        </p>
      </div>
    </div>
  </footer>
     
</div>
`;
